app/Http/Controllers/Backend/StudentController.php

```diff
@@ -68,4 +71,183 @@ public function all(): JsonResponse
+     *                 @OA\Property(property="cv_link", type="string", example="https://cv.pstuian.com/cv_1802043.pdf", description="Student CV URL"),
+     *                 @OA\Property(property="bio", type="string", example="student details", description="Student details"),
+     *                 @OA\Property(property="linkedin", type="string", example="https://www.linkedin.com/in/engg-shishir/", description="LinkedIn account URL"),
+     *                 @OA\Property(property="facebook", type="string", example="https://www.facebook.com/Engg.Shishir/", description="Facebook account URL")
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(response=201, description="Student updated successfully"),
+     *     @OA\Response(response=400, description="Bad request"),
+     *     @OA\Response(response=404, description="Resource Not Found")
+     * )
+     */
+    public function update(StudentRequest $request, int $id): JsonResponse
+    {
+        try {
+            return $this->responseJson(
+                $this->studentService->update($id, $request->all()),
+                Response::HTTP_OK,
+                __('Student updated successfully.')
+            );
+        } catch (Exception $exception) {
+            return $this->responseErrorJson($exception);
+        }
+    }
+
+    /**
+     * @OA\Delete(
+     *      path="/api/v1/backend/students/{id}",
+     *      tags={"Students"},
+     *      summary="Delete a student",
+     *      description="Delete a student by ID",
+     *      @OA\Parameter(
+     *          name="id",
+     *          in="path",
+     *          required=true,
+     *          description="ID of the student to delete",
+     *          @OA\Schema(
+     *              type="integer",
+     *              format="int64"
+     *          )
+     *      ),
+     *      @OA\Parameter(name="deleted", description="Delete type, Soft=1, Hard=9, Keep empty for permanent delete", example="1", required=false, in="query", @OA\Schema(type="integer")),
+     *      @OA\Response(
+     *          response=200,
+     *          description="student deleted successful",
+     *      ),
+     *      @OA\Response(
+     *          response=404,
+     *          description="student not found"
+     *      ),
+     *      @OA\Response(
+     *          response=500,
+     *          description="Internal server error"
+     *      )
+     * )
+     */
+    public function destroy(int $id, Request $request): JsonResponse
+    {
+        try {
+            $deleteStatusInt = (int) $request->deleted ?? DeleteStatus::SOFT_DELETE;
+
+            return $this->responseJson(
+                $this->studentService->delete(
+                    $id,
+                    $deleteStatusInt
+                ),
+                Response::HTTP_OK,
+                __('student deleted state successfully update.')
```

Comment:
I don't like this method.
Cause, if we show this method to frontend, it would not make sense.

Need to handle this something like - 

1. If soft deleted (1) - Student has been moved to trash.
2. Hard deleted (9) - Student has been permanently deleted.
3. Revert back (0) - Student has been restored.

So, maybe we would create a class which will handle delete message by deleted value.
then call it from here.


Class could be like - 

```php
DeleteMessage {
    private int $deletedStatus;
    private string $entityName;

    construct (int $deletedStatus, string $entityName)
    
    public function format(){
        switch($this->deletedStatus) {
             case 0:
                  return sprintf('%s has been restored successfully.', $this->entityName);

             case 1:
                  return
        }
    }
}

```


And then maybe from StudentController - 

```php
 return $this->responseJson(
                $this->studentService->delete(
                    $id,
                    $deleteStatusInt
                ),
                Response::HTTP_OK,
                (new DeleteMessage($deleteStatusInt, 'Student'))->format()
);
```
Just a pseudo code idea.


Please maintain php-8.1 coding standard always.